### PR TITLE
Data Preview 0.2

### PR DESCRIPTION
**Pull request by joezuntz**
DP0.2 doesn't have a proper shear catalog, but does have HSM shape measurements that we should be able to get something out of. SInce it's not in GCR yet (I think) I've directly pulled it from the parquet files. This was a good exercise too.

At the same time I've moved all the ingestion stages into a new subpacakge to tidy things up a bit.

I've also done some small changes that were needed along the way, the biggest being additions to the lens selector to enable a simple magnitude-limited selection.

### CODE REVIEW

File: examples/dp0.2/config.yml [hunk 1]
```diff
@@ -0,0 +1,133 @@
+global:
+    chunk_rows: 1000000
+    pixelization: healpix
+    nside: 256
+    sparse: True
+    delta_gamma: 1e9 # mock value since this should not be used at all
```
[the remark this answers is not in the document]
Comment (joezuntz) [reply examples/dp0.2/config.yml]: Thanks - yep - I've removed this from the base source selector class and just put it in the subclasses.

File: examples/dp0.2/config.yml [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+TXTwoPoint:    
+    flip_g2: True
+    min_sep: 2.5
+    max_sep: 250.0
+    nbins: 20
+    verbose: 1
+
+
+TXSubsampleRandoms:
+    sample_rate: 0.2
```
[the remark this answers is not in the document]
Comment (joezuntz) [reply examples/dp0.2/config.yml]: This wasn't actually used - I've removed it. Now that facility is in the original randoms class.

File: examples/dp0.2/config.yml [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+
+#TXRealGaussianCovariance:
+
+TXTwoPointPlotsTheory:
+    aliases:
+        twopoint_data_real: summary_statistics_real
```
[the remark this answers is not in the document]
Comment (joezuntz) [reply examples/dp0.2/config.yml]: Added a comment to explain.

File: txpipe/ingest/dp02.py [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+                    if i == 0 and j == 0:
+                        photo_outfile = self.setup_output("photometry_catalog", "photometry", photo_data, n)
+                        shear_outfile = self.setup_output("shear_catalog", "shear", shear_data, n)
+
+                    e1 = s1 + len(photo_data['ra'])
+                    e2 = s2 + len(shear_data['ra'])
+                    self.write_output(photo_outfile, "photometry", photo_data, s1, e1)
+                    self.write_output(shear_outfile, "shear", shear_data, s2, e2)
+                    print(f"Processing chunk {j+1}/{n_chunk} of file {i+1}/{nfile} into rows {s1:,} - {e1:,}")
+                    s1 = e1
```
[the remark this answers is not in the document]
Comment (joezuntz) [reply txpipe/ingest/dp02.py]: Updated to end1 and end2

File: txpipe/utils/conversion.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    return -2.5 * np.log10(flux * 1e-9 / REF_FLUX)
+
+def mag_ab_to_nanojansky(mag):
+    return 1e9 * REF_FLUX * 10 ** (-0.4 * mag)
+
+def nanojansky_err_to_mag_ab(flux, flux_err):
+    return 2.5 / np.log(10) * (flux_err / flux)
+
+
+def moments_to_shear(Ixx, Iyy, Ixy):
```
[the remark this answers is not in the document]
Comment (joezuntz) [reply txpipe/utils/conversion.py:16]: I moved this from the old file, it's just in one place now, here.

File: txpipe/utils/hdf_tools.py [hunk 6]
```diff
@@ -162,3 +165,24 @@ def _write(self):
 
     def finish(self):
         self._write()
+
+
+
+def h5py_shorten(group, name, n):
```
[the remark this answers is not in the document]
Comment (joezuntz) [reply txpipe/utils/hdf_tools.py:171]: Ditto this.

**Comment by joezuntz** [comment]
Thanks - I think I've addressed these! Re-requesting review.

**Review by jpratmarti — APPROVED** [review]

**Pull request merged**